// arc-carousel/dist/arc-carousel.es.js
import rr, { createContext as ve, useContext as q, useState as dr, useRef as nt, useCallback as le, useEffect as ge, forwardRef as D, Children as yr, isValidElement as kr, createElement as Wr, cloneElement as be, Fragment as ot } from "react";
var Gr = { exports: {} }, gr = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var ce;
function at() {
  if (ce)
    return gr;
  ce = 1;
  var e = rr, t = Symbol.for("react.element"), o = Symbol.for("react.fragment"), n = Object.prototype.hasOwnProperty, a = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, c = { key: !0, ref: !0, __self: !0, __source: !0 };
  function i(s, u, v) {
    var p, f = {}, g = null, x = null;
    v !== void 0 && (g = "" + v), u.key !== void 0 && (g = "" + u.key), u.ref !== void 0 && (x = u.ref);
    for (p in u)
      n.call(u, p) && !c.hasOwnProperty(p) && (f[p] = u[p]);
    if (s && s.defaultProps)
      for (p in u = s.defaultProps, u)
        f[p] === void 0 && (f[p] = u[p]);
    return { $$typeof: t, type: s, key: g, ref: x, props: f, _owner: a.current };
  }
  return gr.Fragment = o, gr.jsx = i, gr.jsxs = i, gr;
}
var br = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var ue;
function it() {
  return ue || (ue = 1, process.env.NODE_ENV !== "production" && function() {
    var e = rr, t = Symbol.for("react.element"), o = Symbol.for("react.portal"), n = Symbol.for("react.fragment"), a = Symbol.for("react.strict_mode"), c = Symbol.for("react.profiler"), i = Symbol.for("react.provider"), s = Symbol.for("react.context"), u = Symbol.for("react.forward_ref"), v = Symbol.for("react.suspense"), p = Symbol.for("react.suspense_list"), f = Symbol.for("react.memo"), g = Symbol.for("react.lazy"), x = Symbol.for("react.offscreen"), R = Symbol.iterator, T = "@@iterator";
    function A(r) {
      if (r === null || typeof r != "object")
        return null;
      var l = R && r[R] || r[T];
      return typeof l == "function" ? l : null;
    }
    var k = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function m(r) {
      {
        for (var l = arguments.length, d = new Array(l > 1 ? l - 1 : 0), b = 1; b < l; b++)
          d[b - 1] = arguments[b];
        _("error", r, d);
      }
    }
    function _(r, l, d) {
      {
        var b = k.ReactDebugCurrentFrame, C = b.getStackAddendum();
        C !== "" && (l += "%s", d = d.concat([C]));
        var P = d.map(function(w) {
          return String(w);
        });
        P.unshift("Warning: " + l), Function.prototype.apply.call(console[r], console, P);
      }
    }
    var M = !1, N = !1, L = !1, J = !1, Y = !1, $;
    $ = Symbol.for("react.module.reference");
    function er(r) {
      return !!(typeof r == "string" || typeof r == "function" || r === n || r === c || Y || r === a || r === v || r === p || J || r === x || M || N || L || typeof r == "object" && r !== null && (r.$$typeof === g || r.$$typeof === f || r.$$typeof === i || r.$$typeof === s || r.$$typeof === u || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      r.$$typeof === $ || r.getModuleId !== void 0));
    }
    function tr(r, l, d) {
      var b = r.displayName;
      if (b)
        return b;
      var C = l.displayName || l.name || "";
      return C !== "" ? d + "(" + C + ")" : d;
    }
    function S(r) {
      return r.displayName || "Context";
    }
    function z(r) {
      if (r == null)
        return null;
      if (typeof r.tag == "number" && m("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof r == "function")
        return r.displayName || r.name || null;
      if (typeof r == "string")
        return r;
      switch (r) {
        case n:
          return "Fragment";
        case o:
          return "Portal";
        case c:
          return "Profiler";
        case a:
          return "StrictMode";
        case v:
          return "Suspense";
        case p:
          return "SuspenseList";
      }
      if (typeof r == "object")
        switch (r.$$typeof) {
          case s:
            var l = r;
            return S(l) + ".Consumer";
          case i:
            var d = r;
            return S(d._context) + ".Provider";
          case u:
            return tr(r, r.render, "ForwardRef");
          case f:
            var b = r.displayName || null;
            return b !== null ? b : z(r.type) || "Memo";
          case g: {
            var C = r, P = C._payload, w = C._init;
            try {
              return z(w(P));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var F = Object.assign, X = 0, nr, fr, lr, Z, pr, K, or;
    function W() {
    }
    W.__reactDisabledLog = !0;
    function Oe() {
      {
        if (X === 0) {
          nr = console.log, fr = console.info, lr = console.warn, Z = console.error, pr = console.group, K = console.groupCollapsed, or = console.groupEnd;
          var r = {
            configurable: !0,
            enumerable: !0,
            value: W,
            writable: !0
          };
          Object.defineProperties(console, {
            info: r,
            log: r,
            warn: r,
            error: r,
            group: r,
            groupCollapsed: r,
            groupEnd: r
          });
        }
        X++;
      }
    }
    function Ie() {
      {
        if (X--, X === 0) {
          var r = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: F({}, r, {
              value: nr
            }),
            info: F({}, r, {
              value: fr
            }),
            warn: F({}, r, {
              value: lr
            }),
            error: F({}, r, {
              value: Z
            }),
            group: F({}, r, {
              value: pr
            }),
            groupCollapsed: F({}, r, {
              value: K
            }),
            groupEnd: F({}, r, {
              value: or
            })
          });
        }
        X < 0 && m("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var jr = k.ReactCurrentDispatcher, Ar;
    function wr(r, l, d) {
      {
        if (Ar === void 0)
          try {
            throw Error();
          } catch (C) {
            var b = C.stack.trim().match(/\n( *(at )?)/);
            Ar = b && b[1] || "";
          }
        return `
` + Ar + r;
      }
    }
    var Or = !1, Cr;
    {
      var Me = typeof WeakMap == "function" ? WeakMap : Map;
      Cr = new Me();
    }
    function Hr(r, l) {
      if (!r || Or)
        return "";
      {
        var d = Cr.get(r);
        if (d !== void 0)
          return d;
      }
      var b;
      Or = !0;
      var C = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var P;
      P = jr.current, jr.current = null, Oe();
      try {
        if (l) {
          var w = function() {
            throw Error();
          };
          if (Object.defineProperty(w.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(w, []);
            } catch (U) {
              b = U;
            }
            Reflect.construct(r, [], w);
          } else {
            try {
              w.call();
            } catch (U) {
              b = U;
            }
            r.call(w.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (U) {
            b = U;
          }
          r();
        }
      } catch (U) {
        if (U && b && typeof U.stack == "string") {
          for (var h = U.stack.split(`
`), G = b.stack.split(`
`), O = h.length - 1, I = G.length - 1; O >= 1 && I >= 0 && h[O] !== G[I]; )
            I--;
          for (; O >= 1 && I >= 0; O--, I--)
            if (h[O] !== G[I]) {
              if (O !== 1 || I !== 1)
                do
                  if (O--, I--, I < 0 || h[O] !== G[I]) {
                    var B = `
` + h[O].replace(" at new ", " at ");
                    return r.displayName && B.includes("<anonymous>") && (B = B.replace("<anonymous>", r.displayName)), typeof r == "function" && Cr.set(r, B), B;
                  }
                while (O >= 1 && I >= 0);
              break;
            }
        }
      } finally {
        Or = !1, jr.current = P, Ie(), Error.prepareStackTrace = C;
      }
      var ur = r ? r.displayName || r.name : "", se = ur ? wr(ur) : "";
      return typeof r == "function" && Cr.set(r, se), se;
    }
    function $e(r, l, d) {
      return Hr(r, !1);
    }
    function ze(r) {
      var l = r.prototype;
      return !!(l && l.isReactComponent);
    }
    function Er(r, l, d) {
      if (r == null)
        return "";
      if (typeof r == "function")
        return Hr(r, ze(r));
      if (typeof r == "string")
        return wr(r);
      switch (r) {
        case v:
          return wr("Suspense");
        case p:
          return wr("SuspenseList");
      }
      if (typeof r == "object")
        switch (r.$$typeof) {
          case u:
            return $e(r.render);
          case f:
            return Er(r.type, l, d);
          case g: {
            var b = r, C = b._payload, P = b._init;
            try {
              return Er(P(C), l, d);
            } catch {
            }
          }
        }
      return "";
    }
    var Rr = Object.prototype.hasOwnProperty, Jr = {}, Xr = k.ReactDebugCurrentFrame;
    function Sr(r) {
      if (r) {
        var l = r._owner, d = Er(r.type, r._source, l ? l.type : null);
        Xr.setExtraStackFrame(d);
      } else
        Xr.setExtraStackFrame(null);
    }
    function Ne(r, l, d, b, C) {
      {
        var P = Function.call.bind(Rr);
        for (var w in r)
          if (P(r, w)) {
            var h = void 0;
            try {
              if (typeof r[w] != "function") {
                var G = Error((b || "React class") + ": " + d + " type `" + w + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof r[w] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw G.name = "Invariant Violation", G;
              }
              h = r[w](l, w, b, d, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (O) {
              h = O;
            }
            h && !(h instanceof Error) && (Sr(C), m("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", b || "React class", d, w, typeof h), Sr(null)), h instanceof Error && !(h.message in Jr) && (Jr[h.message] = !0, Sr(C), m("Failed %s type: %s", d, h.message), Sr(null));
          }
      }
    }
    var We = Array.isArray;
    function Ir(r) {
      return We(r);
    }
    function Ge(r) {
      {
        var l = typeof Symbol == "function" && Symbol.toStringTag, d = l && r[Symbol.toStringTag] || r.constructor.name || "Object";
        return d;
      }
    }
    function Le(r) {
      try {
        return Zr(r), !1;
      } catch {
        return !0;
      }
    }
    function Zr(r) {
      return "" + r;
    }
    function Kr(r) {
      if (Le(r))
        return m("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", Ge(r)), Zr(r);
    }
    var vr = k.ReactCurrentOwner, Fe = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, Qr, re, Mr;
    Mr = {};
    function Be(r) {
      if (Rr.call(r, "ref")) {
        var l = Object.getOwnPropertyDescriptor(r, "ref").get;
        if (l && l.isReactWarning)
          return !1;
      }
      return r.ref !== void 0;
    }
    function De(r) {
      if (Rr.call(r, "key")) {
        var l = Object.getOwnPropertyDescriptor(r, "key").get;
        if (l && l.isReactWarning)
          return !1;
      }
      return r.key !== void 0;
    }
    function Ve(r, l) {
      if (typeof r.ref == "string" && vr.current && l && vr.current.stateNode !== l) {
        var d = z(vr.current.type);
        Mr[d] || (m('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', z(vr.current.type), r.ref), Mr[d] = !0);
      }
    }
    function Ye(r, l) {
      {
        var d = function() {
          Qr || (Qr = !0, m("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", l));
        };
        d.isReactWarning = !0, Object.defineProperty(r, "key", {
          get: d,
          configurable: !0
        });
      }
    }
    function Ue(r, l) {
      {
        var d = function() {
          re || (re = !0, m("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", l));
        };
        d.isReactWarning = !0, Object.defineProperty(r, "ref", {
          get: d,
          configurable: !0
        });
      }
    }
    var qe = function(r, l, d, b, C, P, w) {
      var h = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: t,
        // Built-in properties that belong on the element
        type: r,
        key: l,
        ref: d,
        props: w,
        // Record the component responsible for creating this element.
        _owner: P
      };
      return h._store = {}, Object.defineProperty(h._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(h, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: b
      }), Object.defineProperty(h, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: C
      }), Object.freeze && (Object.freeze(h.props), Object.freeze(h)), h;
    };
    function He(r, l, d, b, C) {
      {
        var P, w = {}, h = null, G = null;
        d !== void 0 && (Kr(d), h = "" + d), De(l) && (Kr(l.key), h = "" + l.key), Be(l) && (G = l.ref, Ve(l, C));
        for (P in l)
          Rr.call(l, P) && !Fe.hasOwnProperty(P) && (w[P] = l[P]);
        if (r && r.defaultProps) {
          var O = r.defaultProps;
          for (P in O)
            w[P] === void 0 && (w[P] = O[P]);
        }
        if (h || G) {
          var I = typeof r == "function" ? r.displayName || r.name || "Unknown" : r;
          h && Ye(w, I), G && Ue(w, I);
        }
        return qe(r, h, G, C, b, vr.current, w);
      }
    }
    var $r = k.ReactCurrentOwner, ee = k.ReactDebugCurrentFrame;
    function cr(r) {
      if (r) {
        var l = r._owner, d = Er(r.type, r._source, l ? l.type : null);
        ee.setExtraStackFrame(d);
      } else
        ee.setExtraStackFrame(null);
    }
    var zr;
    zr = !1;
    function Nr(r) {
      return typeof r == "object" && r !== null && r.$$typeof === t;
    }
    function te() {
      {
        if ($r.current) {
          var r = z($r.current.type);
          if (r)
            return `

Check the render method of \`` + r + "`.";
        }
        return "";
      }
    }
    function Je(r) {
      {
        if (r !== void 0) {
          var l = r.fileName.replace(/^.*[\\\/]/, ""), d = r.lineNumber;
          return `

Check your code at ` + l + ":" + d + ".";
        }
        return "";
      }
    }
    var ne = {};
    function Xe(r) {
      {
        var l = te();
        if (!l) {
          var d = typeof r == "string" ? r : r.displayName || r.name;
          d && (l = `

Check the top-level render call using <` + d + ">.");
        }
        return l;
      }
    }
    function oe(r, l) {
      {
        if (!r._store || r._store.validated || r.key != null)
          return;
        r._store.validated = !0;
        var d = Xe(l);
        if (ne[d])
          return;
        ne[d] = !0;
        var b = "";
        r && r._owner && r._owner !== $r.current && (b = " It was passed a child from " + z(r._owner.type) + "."), cr(r), m('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', d, b), cr(null);
      }
    }
    function ae(r, l) {
      {
        if (typeof r != "object")
          return;
        if (Ir(r))
          for (var d = 0; d < r.length; d++) {
            var b = r[d];
            Nr(b) && oe(b, l);
          }
        else if (Nr(r))
          r._store && (r._store.validated = !0);
        else if (r) {
          var C = A(r);
          if (typeof C == "function" && C !== r.entries)
            for (var P = C.call(r), w; !(w = P.next()).done; )
              Nr(w.value) && oe(w.value, l);
        }
      }
    }
    function Ze(r) {
      {
        var l = r.type;
        if (l == null || typeof l == "string")
          return;
        var d;
        if (typeof l == "function")
          d = l.propTypes;
        else if (typeof l == "object" && (l.$$typeof === u || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        l.$$typeof === f))
          d = l.propTypes;
        else
          return;
        if (d) {
          var b = z(l);
          Ne(d, r.props, "prop", b, r);
        } else if (l.PropTypes !== void 0 && !zr) {
          zr = !0;
          var C = z(l);
          m("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", C || "Unknown");
        }
        typeof l.getDefaultProps == "function" && !l.getDefaultProps.isReactClassApproved && m("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function Ke(r) {
      {
        for (var l = Object.keys(r.props), d = 0; d < l.length; d++) {
          var b = l[d];
          if (b !== "children" && b !== "key") {
            cr(r), m("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", b), cr(null);
            break;
          }
        }
        r.ref !== null && (cr(r), m("Invalid attribute `ref` supplied to `React.Fragment`."), cr(null));
      }
    }
    function ie(r, l, d, b, C, P) {
      {
        var w = er(r);
        if (!w) {
          var h = "";
          (r === void 0 || typeof r == "object" && r !== null && Object.keys(r).length === 0) && (h += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var G = Je(C);
          G ? h += G : h += te();
          var O;
          r === null ? O = "null" : Ir(r) ? O = "array" : r !== void 0 && r.$$typeof === t ? (O = "<" + (z(r.type) || "Unknown") + " />", h = " Did you accidentally export a JSX literal instead of a component?") : O = typeof r, m("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", O, h);
        }
        var I = He(r, l, d, C, P);
        if (I == null)
          return I;
        if (w) {
          var B = l.children;
          if (B !== void 0)
            if (b)
              if (Ir(B)) {
                for (var ur = 0; ur < B.length; ur++)
                  ae(B[ur], r);
                Object.freeze && Object.freeze(B);
              } else
                m("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              ae(B, r);
        }
        return r === n ? Ke(I) : Ze(I), I;
      }
    }
    function Qe(r, l, d) {
      return ie(r, l, d, !0);
    }
    function rt(r, l, d) {
      return ie(r, l, d, !1);
    }
    var et = rt, tt = Qe;
    br.Fragment = n, br.jsx = et, br.jsxs = tt;
  }()), br;
}
process.env.NODE_ENV === "production" ? Gr.exports = at() : Gr.exports = it();
var E = Gr.exports;
const me = ve({}), st = ({ children: e, ...t }) => /* @__PURE__ */ E.jsx(me.Provider, { value: { ...t }, children: e }), Dr = () => q(me), H = ve({}), he = ({
  children: e,
  orientation: t,
  slideCount: o,
  pagesCount: n,
  slidesPerPage: a,
  initialPage: c,
  hasLoop: i,
  transition: s,
  transitionDuration: u,
  gap: v,
  autoplay: p,
  autoplayInterval: f
}) => {
  const [g, x] = dr(() => c < 1 ? 0 : c > n ? n - 1 : c - 1), { onPageChange: R } = Dr(), T = () => {
    const Y = g;
    let $ = 0;
    i ? $ = (g + 1) % n : $ = g === n - 1 ? g : g + 1, R == null || R(Y, $), x($);
  }, A = () => {
    const Y = g;
    let $ = 0;
    i ? $ = (g - 1 + n) % n : $ = g === 0 ? g : g - 1, R == null || R(Y, $), x($);
  }, [k, m] = dr(p), [_, M] = dr(!1), N = nt(), L = le(() => {
    m(!0);
  }, []), J = le(() => {
    m(!1);
  }, []);
  return ge(() => (k ? N.current = setInterval(() => {
    T();
  }, f) : clearInterval(N.current), () => {
    clearInterval(N.current);
  }), [k, T]), /* @__PURE__ */ E.jsx(
    H.Provider,
    {
      value: {
        orientation: t,
        currentPage: g,
        setCurrentPage: x,
        next: T,
        prev: A,
        slideCount: o,
        pagesCount: n,
        slidesPerPage: a,
        initialPage: c,
        hasLoop: i,
        transition: s,
        transitionDuration: u,
        gap: v,
        autoplay: p,
        startAutoplay: L,
        stopAutoplay: J,
        isAutoplaying: k,
        isPaused: _,
        setIsPaused: M
      },
      children: e
    }
  );
};
he.displayName = "CarouselProvider";
function ye(e) {
  var t, o, n = "";
  if (typeof e == "string" || typeof e == "number")
    n += e;
  else if (typeof e == "object")
    if (Array.isArray(e))
      for (t = 0; t < e.length; t++)
        e[t] && (o = ye(e[t])) && (n && (n += " "), n += o);
    else
      for (t in e)
        e[t] && (n && (n += " "), n += t);
  return n;
}
function lt() {
  for (var e, t, o = 0, n = ""; o < arguments.length; )
    (e = arguments[o++]) && (t = ye(e)) && (n && (n += " "), n += t);
  return n;
}
function ct() {
  for (var e = 0, t, o, n = ""; e < arguments.length; )
    (t = arguments[e++]) && (o = xe(t)) && (n && (n += " "), n += o);
  return n;
}
function xe(e) {
  if (typeof e == "string")
    return e;
  for (var t, o = "", n = 0; n < e.length; n++)
    e[n] && (t = xe(e[n])) && (o && (o += " "), o += t);
  return o;
}
var Vr = "-";
function ut(e) {
  var t = ft(e), o = e.conflictingClassGroups, n = e.conflictingClassGroupModifiers, a = n === void 0 ? {} : n;
  function c(s) {
    var u = s.split(Vr);
    return u[0] === "" && u.length !== 1 && u.shift(), we(u, t) || dt(s);
  }
  function i(s, u) {
    var v = o[s] || [];
    return u && a[s] ? [].concat(v, a[s]) : v;
  }
  return {
    getClassGroupId: c,
    getConflictingClassGroupIds: i
  };
}
function we(e, t) {
  var i;
  if (e.length === 0)
    return t.classGroupId;
  var o = e[0], n = t.nextPart.get(o), a = n ? we(e.slice(1), n) : void 0;
  if (a)
    return a;
  if (t.validators.length !== 0) {
    var c = e.join(Vr);
    return (i = t.validators.find(function(s) {
      var u = s.validator;
      return u(c);
    })) == null ? void 0 : i.classGroupId;
  }
}
var de = /^\[(.+)\]$/;
function dt(e) {
  if (de.test(e)) {
    var t = de.exec(e)[1], o = t == null ? void 0 : t.substring(0, t.indexOf(":"));
    if (o)
      return "arbitrary.." + o;
  }
}
function ft(e) {
  var t = e.theme, o = e.prefix, n = {
    nextPart: /* @__PURE__ */ new Map(),
    validators: []
  }, a = vt(Object.entries(e.classGroups), o);
  return a.forEach(function(c) {
    var i = c[0], s = c[1];
    Lr(s, n, i, t);
  }), n;
}
function Lr(e, t, o, n) {
  e.forEach(function(a) {
    if (typeof a == "string") {
      var c = a === "" ? t : fe(t, a);
      c.classGroupId = o;
      return;
    }
    if (typeof a == "function") {
      if (pt(a)) {
        Lr(a(n), t, o, n);
        return;
      }
      t.validators.push({
        validator: a,
        classGroupId: o
      });
      return;
    }
    Object.entries(a).forEach(function(i) {
      var s = i[0], u = i[1];
      Lr(u, fe(t, s), o, n);
    });
  });
}
function fe(e, t) {
  var o = e;
  return t.split(Vr).forEach(function(n) {
    o.nextPart.has(n) || o.nextPart.set(n, {
      nextPart: /* @__PURE__ */ new Map(),
      validators: []
    }), o = o.nextPart.get(n);
  }), o;
}
function pt(e) {
  return e.isThemeGetter;
}
function vt(e, t) {
  return t ? e.map(function(o) {
    var n = o[0], a = o[1], c = a.map(function(i) {
      return typeof i == "string" ? t + i : typeof i == "object" ? Object.fromEntries(Object.entries(i).map(function(s) {
        var u = s[0], v = s[1];
        return [t + u, v];
      })) : i;
    });
    return [n, c];
  }) : e;
}
function gt(e) {
  if (e < 1)
    return {
      get: function() {
      },
      set: function() {
      }
    };
  var t = 0, o = /* @__PURE__ */ new Map(), n = /* @__PURE__ */ new Map();
  function a(c, i) {
    o.set(c, i), t++, t > e && (t = 0, n = o, o = /* @__PURE__ */ new Map());
  }
  return {
    get: function(i) {
      var s = o.get(i);
      if (s !== void 0)
        return s;
      if ((s = n.get(i)) !== void 0)
        return a(i, s), s;
    },
    set: function(i, s) {
      o.has(i) ? o.set(i, s) : a(i, s);
    }
  };
}
var Ce = "!";
function bt(e) {
  var t = e.separator || ":", o = t.length === 1, n = t[0], a = t.length;
  return function(i) {
    for (var s = [], u = 0, v = 0, p, f = 0; f < i.length; f++) {
      var g = i[f];
      if (u === 0) {
        if (g === n && (o || i.slice(f, f + a) === t)) {
          s.push(i.slice(v, f)), v = f + a;
          continue;
        }
        if (g === "/") {
          p = f;
          continue;
        }
      }
      g === "[" ? u++ : g === "]" && u--;
    }
    var x = s.length === 0 ? i : i.substring(v), R = x.startsWith(Ce), T = R ? x.substring(1) : x, A = p && p > v ? p - v : void 0;
    return {
      modifiers: s,
      hasImportantModifier: R,
      baseClassName: T,
      maybePostfixModifierPosition: A
    };
  };
}
function mt(e) {
  if (e.length <= 1)
    return e;
  var t = [], o = [];
  return e.forEach(function(n) {
    var a = n[0] === "[";
    a ? (t.push.apply(t, o.sort().concat([n])), o = []) : o.push(n);
  }), t.push.apply(t, o.sort()), t;
}
function ht(e) {
  return {
    cache: gt(e.cacheSize),
    splitModifiers: bt(e),
    ...ut(e)
  };
}
var yt = /\s+/;
function xt(e, t) {
  var o = t.splitModifiers, n = t.getClassGroupId, a = t.getConflictingClassGroupIds, c = /* @__PURE__ */ new Set();
  return e.trim().split(yt).map(function(i) {
    var s = o(i), u = s.modifiers, v = s.hasImportantModifier, p = s.baseClassName, f = s.maybePostfixModifierPosition, g = n(f ? p.substring(0, f) : p), x = !!f;
    if (!g) {
      if (!f)
        return {
          isTailwindClass: !1,
          originalClassName: i
        };
      if (g = n(p), !g)
        return {
          isTailwindClass: !1,
          originalClassName: i
        };
      x = !1;
    }
    var R = mt(u).join(":"), T = v ? R + Ce : R;
    return {
      isTailwindClass: !0,
      modifierId: T,
      classGroupId: g,
      originalClassName: i,
      hasPostfixModifier: x
    };
  }).reverse().filter(function(i) {
    if (!i.isTailwindClass)
      return !0;
    var s = i.modifierId, u = i.classGroupId, v = i.hasPostfixModifier, p = s + u;
    return c.has(p) ? !1 : (c.add(p), a(u, v).forEach(function(f) {
      return c.add(s + f);
    }), !0);
  }).reverse().map(function(i) {
    return i.originalClassName;
  }).join(" ");
}
function wt() {
  for (var e = arguments.length, t = new Array(e), o = 0; o < e; o++)
    t[o] = arguments[o];
  var n, a, c, i = s;
  function s(v) {
    var p = t[0], f = t.slice(1), g = f.reduce(function(x, R) {
      return R(x);
    }, p());
    return n = ht(g), a = n.cache.get, c = n.cache.set, i = u, u(v);
  }
  function u(v) {
    var p = a(v);
    if (p)
      return p;
    var f = xt(v, n);
    return c(v, f), f;
  }
  return function() {
    return i(ct.apply(null, arguments));
  };
}
function j(e) {
  var t = function(n) {
    return n[e] || [];
  };
  return t.isThemeGetter = !0, t;
}
var Ee = /^\[(?:([a-z-]+):)?(.+)\]$/i, Ct = /^\d+\/\d+$/, Et = /* @__PURE__ */ new Set(["px", "full", "screen"]), Rt = /^(\d+(\.\d+)?)?(xs|sm|md|lg|xl)$/, St = /\d+(%|px|r?em|[sdl]?v([hwib]|min|max)|pt|pc|in|cm|mm|cap|ch|ex|r?lh|cq(w|h|i|b|min|max))|\b(calc|min|max|clamp)\(.+\)|^0$/, Tt = /^-?((\d+)?\.?(\d+)[a-z]+|0)_-?((\d+)?\.?(\d+)[a-z]+|0)/;
function V(e) {
  return ar(e) || Et.has(e) || Ct.test(e) || Fr(e);
}
function Fr(e) {
  return ir(e, "length", Ot);
}
function kt(e) {
  return ir(e, "size", Re);
}
function Pt(e) {
  return ir(e, "position", Re);
}
function _t(e) {
  return ir(e, "url", It);
}
function Tr(e) {
  return ir(e, "number", ar);
}
function ar(e) {
  return !Number.isNaN(Number(e));
}
function jt(e) {
  return e.endsWith("%") && ar(e.slice(0, -1));
}
function mr(e) {
  return pe(e) || ir(e, "number", pe);
}
function y(e) {
  return Ee.test(e);
}
function hr() {
  return !0;
}
function Q(e) {
  return Rt.test(e);
}
function At(e) {
  return ir(e, "", Mt);
}
function ir(e, t, o) {
  var n = Ee.exec(e);
  return n ? n[1] ? n[1] === t : o(n[2]) : !1;
}
function Ot(e) {
  return St.test(e);
}
function Re() {
  return !1;
}
function It(e) {
  return e.startsWith("url(");
}
function pe(e) {
  return Number.isInteger(Number(e));
}
function Mt(e) {
  return Tt.test(e);
}
function $t() {
  var e = j("colors"), t = j("spacing"), o = j("blur"), n = j("brightness"), a = j("borderColor"), c = j("borderRadius"), i = j("borderSpacing"), s = j("borderWidth"), u = j("contrast"), v = j("grayscale"), p = j("hueRotate"), f = j("invert"), g = j("gap"), x = j("gradientColorStops"), R = j("gradientColorStopPositions"), T = j("inset"), A = j("margin"), k = j("opacity"), m = j("padding"), _ = j("saturate"), M = j("scale"), N = j("sepia"), L = j("skew"), J = j("space"), Y = j("translate"), $ = function() {
    return ["auto", "contain", "none"];
  }, er = function() {
    return ["auto", "hidden", "clip", "visible", "scroll"];
  }, tr = function() {
    return ["auto", y, t];
  }, S = function() {
    return [y, t];
  }, z = function() {
    return ["", V];
  }, F = function() {
    return ["auto", ar, y];
  }, X = function() {
    return ["bottom", "center", "left", "left-bottom", "left-top", "right", "right-bottom", "right-top", "top"];
  }, nr = function() {
    return ["solid", "dashed", "dotted", "double", "none"];
  }, fr = function() {
    return ["normal", "multiply", "screen", "overlay", "darken", "lighten", "color-dodge", "color-burn", "hard-light", "soft-light", "difference", "exclusion", "hue", "saturation", "color", "luminosity", "plus-lighter"];
  }, lr = function() {
    return ["start", "end", "center", "between", "around", "evenly", "stretch"];
  }, Z = function() {
    return ["", "0", y];
  }, pr = function() {
    return ["auto", "avoid", "all", "avoid-page", "page", "left", "right", "column"];
  }, K = function() {
    return [ar, Tr];
  }, or = function() {
    return [ar, y];
  };
  return {
    cacheSize: 500,
    theme: {
      colors: [hr],
      spacing: [V],
      blur: ["none", "", Q, y],
      brightness: K(),
      borderColor: [e],
      borderRadius: ["none", "", "full", Q, y],
      borderSpacing: S(),
      borderWidth: z(),
      contrast: K(),
      grayscale: Z(),
      hueRotate: or(),
      invert: Z(),
      gap: S(),
      gradientColorStops: [e],
      gradientColorStopPositions: [jt, Fr],
      inset: tr(),
      margin: tr(),
      opacity: K(),
      padding: S(),
      saturate: K(),
      scale: K(),
      sepia: Z(),
      skew: or(),
      space: S(),
      translate: S()
    },
    classGroups: {
      // Layout
      /**
       * Aspect Ratio
       * @see https://tailwindcss.com/docs/aspect-ratio
       */
      aspect: [{
        aspect: ["auto", "square", "video", y]
      }],
      /**
       * Container
       * @see https://tailwindcss.com/docs/container
       */
      container: ["container"],
      /**
       * Columns
       * @see https://tailwindcss.com/docs/columns
       */
      columns: [{
        columns: [Q]
      }],
      /**
       * Break After
       * @see https://tailwindcss.com/docs/break-after
       */
      "break-after": [{
        "break-after": pr()
      }],
      /**
       * Break Before
       * @see https://tailwindcss.com/docs/break-before
       */
      "break-before": [{
        "break-before": pr()
      }],
      /**
       * Break Inside
       * @see https://tailwindcss.com/docs/break-inside
       */
      "break-inside": [{
        "break-inside": ["auto", "avoid", "avoid-page", "avoid-column"]
      }],
      /**
       * Box Decoration Break
       * @see https://tailwindcss.com/docs/box-decoration-break
       */
      "box-decoration": [{
        "box-decoration": ["slice", "clone"]
      }],
      /**
       * Box Sizing
       * @see https://tailwindcss.com/docs/box-sizing
       */
      box: [{
        box: ["border", "content"]
      }],
      /**
       * Display
       * @see https://tailwindcss.com/docs/display
       */
      display: ["block", "inline-block", "inline", "flex", "inline-flex", "table", "inline-table", "table-caption", "table-cell", "table-column", "table-column-group", "table-footer-group", "table-header-group", "table-row-group", "table-row", "flow-root", "grid", "inline-grid", "contents", "list-item", "hidden"],
      /**
       * Floats
       * @see https://tailwindcss.com/docs/float
       */
      float: [{
        float: ["right", "left", "none"]
      }],
      /**
       * Clear
       * @see https://tailwindcss.com/docs/clear
       */
      clear: [{
        clear: ["left", "right", "both", "none"]
      }],
      /**
       * Isolation
       * @see https://tailwindcss.com/docs/isolation
       */
      isolation: ["isolate", "isolation-auto"],
      /**
       * Object Fit
       * @see https://tailwindcss.com/docs/object-fit
       */
      "object-fit": [{
        object: ["contain", "cover", "fill", "none", "scale-down"]
      }],
      /**
       * Object Position
       * @see https://tailwindcss.com/docs/object-position
       */
      "object-position": [{
        object: [].concat(X(), [y])
      }],
      /**
       * Overflow
       * @see https://tailwindcss.com/docs/overflow
       */
      overflow: [{
        overflow: er()
      }],
      /**
       * Overflow X
       * @see https://tailwindcss.com/docs/overflow
       */
      "overflow-x": [{
        "overflow-x": er()
      }],
      /**
       * Overflow Y
       * @see https://tailwindcss.com/docs/overflow
       */
      "overflow-y": [{
        "overflow-y": er()
      }],
      /**
       * Overscroll Behavior
       * @see https://tailwindcss.com/docs/overscroll-behavior
       */
      overscroll: [{
        overscroll: $()
      }],
      /**
       * Overscroll Behavior X
       * @see https://tailwindcss.com/docs/overscroll-behavior
       */
      "overscroll-x": [{
        "overscroll-x": $()
      }],
      /**
       * Overscroll Behavior Y
       * @see https://tailwindcss.com/docs/overscroll-behavior
       */
      "overscroll-y": [{
        "overscroll-y": $()
      }],
      /**
       * Position
       * @see https://tailwindcss.com/docs/position
       */
      position: ["static", "fixed", "absolute", "relative", "sticky"],
      /**
       * Top / Right / Bottom / Left
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      inset: [{
        inset: [T]
      }],
      /**
       * Right / Left
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      "inset-x": [{
        "inset-x": [T]
      }],
      /**
       * Top / Bottom
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      "inset-y": [{
        "inset-y": [T]
      }],
      /**
       * Start
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      start: [{
        start: [T]
      }],
      /**
       * End
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      end: [{
        end: [T]
      }],
      /**
       * Top
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      top: [{
        top: [T]
      }],
      /**
       * Right
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      right: [{
        right: [T]
      }],
      /**
       * Bottom
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      bottom: [{
        bottom: [T]
      }],
      /**
       * Left
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      left: [{
        left: [T]
      }],
      /**
       * Visibility
       * @see https://tailwindcss.com/docs/visibility
       */
      visibility: ["visible", "invisible", "collapse"],
      /**
       * Z-Index
       * @see https://tailwindcss.com/docs/z-index
       */
      z: [{
        z: ["auto", mr]
      }],
      // Flexbox and Grid
      /**
       * Flex Basis
       * @see https://tailwindcss.com/docs/flex-basis
       */
      basis: [{
        basis: tr()
      }],
      /**
       * Flex Direction
       * @see https://tailwindcss.com/docs/flex-direction
       */
      "flex-direction": [{
        flex: ["row", "row-reverse", "col", "col-reverse"]
      }],
      /**
       * Flex Wrap
       * @see https://tailwindcss.com/docs/flex-wrap
       */
      "flex-wrap": [{
        flex: ["wrap", "wrap-reverse", "nowrap"]
      }],
      /**
       * Flex
       * @see https://tailwindcss.com/docs/flex
       */
      flex: [{
        flex: ["1", "auto", "initial", "none", y]
      }],
      /**
       * Flex Grow
       * @see https://tailwindcss.com/docs/flex-grow
       */
      grow: [{
        grow: Z()
      }],
      /**
       * Flex Shrink
       * @see https://tailwindcss.com/docs/flex-shrink
       */
      shrink: [{
        shrink: Z()
      }],
      /**
       * Order
       * @see https://tailwindcss.com/docs/order
       */
      order: [{
        order: ["first", "last", "none", mr]
      }],
      /**
       * Grid Template Columns
       * @see https://tailwindcss.com/docs/grid-template-columns
       */
      "grid-cols": [{
        "grid-cols": [hr]
      }],
      /**
       * Grid Column Start / End
       * @see https://tailwindcss.com/docs/grid-column
       */
      "col-start-end": [{
        col: ["auto", {
          span: ["full", mr]
        }, y]
      }],
      /**
       * Grid Column Start
       * @see https://tailwindcss.com/docs/grid-column
       */
      "col-start": [{
        "col-start": F()
      }],
      /**
       * Grid Column End
       * @see https://tailwindcss.com/docs/grid-column
       */
      "col-end": [{
        "col-end": F()
      }],
      /**
       * Grid Template Rows
       * @see https://tailwindcss.com/docs/grid-template-rows
       */
      "grid-rows": [{
        "grid-rows": [hr]
      }],
      /**
       * Grid Row Start / End
       * @see https://tailwindcss.com/docs/grid-row
       */
      "row-start-end": [{
        row: ["auto", {
          span: [mr]
        }, y]
      }],
      /**
       * Grid Row Start
       * @see https://tailwindcss.com/docs/grid-row
       */
      "row-start": [{
        "row-start": F()
      }],
      /**
       * Grid Row End
       * @see https://tailwindcss.com/docs/grid-row
       */
      "row-end": [{
        "row-end": F()
      }],
      /**
       * Grid Auto Flow
       * @see https://tailwindcss.com/docs/grid-auto-flow
       */
      "grid-flow": [{
        "grid-flow": ["row", "col", "dense", "row-dense", "col-dense"]
      }],
      /**
       * Grid Auto Columns
       * @see https://tailwindcss.com/docs/grid-auto-columns
       */
      "auto-cols": [{
        "auto-cols": ["auto", "min", "max", "fr", y]
      }],
      /**
       * Grid Auto Rows
       * @see https://tailwindcss.com/docs/grid-auto-rows
       */
      "auto-rows": [{
        "auto-rows": ["auto", "min", "max", "fr", y]
      }],
      /**
       * Gap
       * @see https://tailwindcss.com/docs/gap
       */
      gap: [{
        gap: [g]
      }],
      /**
       * Gap X
       * @see https://tailwindcss.com/docs/gap
       */
      "gap-x": [{
        "gap-x": [g]
      }],
      /**
       * Gap Y
       * @see https://tailwindcss.com/docs/gap
       */
      "gap-y": [{
        "gap-y": [g]
      }],
      /**
       * Justify Content
       * @see https://tailwindcss.com/docs/justify-content
       */
      "justify-content": [{
        justify: ["normal"].concat(lr())
      }],
      /**
       * Justify Items
       * @see https://tailwindcss.com/docs/justify-items
       */
      "justify-items": [{
        "justify-items": ["start", "end", "center", "stretch"]
      }],
      /**
       * Justify Self
       * @see https://tailwindcss.com/docs/justify-self
       */
      "justify-self": [{
        "justify-self": ["auto", "start", "end", "center", "stretch"]
      }],
      /**
       * Align Content
       * @see https://tailwindcss.com/docs/align-content
       */
      "align-content": [{
        content: ["normal"].concat(lr(), ["baseline"])
      }],
      /**
       * Align Items
       * @see https://tailwindcss.com/docs/align-items
       */
      "align-items": [{
        items: ["start", "end", "center", "baseline", "stretch"]
      }],
      /**
       * Align Self
       * @see https://tailwindcss.com/docs/align-self
       */
      "align-self": [{
        self: ["auto", "start", "end", "center", "stretch", "baseline"]
      }],
      /**
       * Place Content
       * @see https://tailwindcss.com/docs/place-content
       */
      "place-content": [{
        "place-content": [].concat(lr(), ["baseline"])
      }],
      /**
       * Place Items
       * @see https://tailwindcss.com/docs/place-items
       */
      "place-items": [{
        "place-items": ["start", "end", "center", "baseline", "stretch"]
      }],
      /**
       * Place Self
       * @see https://tailwindcss.com/docs/place-self
       */
      "place-self": [{
        "place-self": ["auto", "start", "end", "center", "stretch"]
      }],
      // Spacing
      /**
       * Padding
       * @see https://tailwindcss.com/docs/padding
       */
      p: [{
        p: [m]
      }],
      /**
       * Padding X
       * @see https://tailwindcss.com/docs/padding
       */
      px: [{
        px: [m]
      }],
      /**
       * Padding Y
       * @see https://tailwindcss.com/docs/padding
       */
      py: [{
        py: [m]
      }],
      /**
       * Padding Start
       * @see https://tailwindcss.com/docs/padding
       */
      ps: [{
        ps: [m]
      }],
      /**
       * Padding End
       * @see https://tailwindcss.com/docs/padding
       */
      pe: [{
        pe: [m]
      }],
      /**
       * Padding Top
       * @see https://tailwindcss.com/docs/padding
       */
      pt: [{
        pt: [m]
      }],
      /**
       * Padding Right
       * @see https://tailwindcss.com/docs/padding
       */
      pr: [{
        pr: [m]
      }],
      /**
       * Padding Bottom
       * @see https://tailwindcss.com/docs/padding
       */
      pb: [{
        pb: [m]
      }],
      /**
       * Padding Left
       * @see https://tailwindcss.com/docs/padding
       */
      pl: [{
        pl: [m]
      }],
      /**
       * Margin
       * @see https://tailwindcss.com/docs/margin
       */
      m: [{
        m: [A]
      }],
      /**
       * Margin X
       * @see https://tailwindcss.com/docs/margin
       */
      mx: [{
        mx: [A]
      }],
      /**
       * Margin Y
       * @see https://tailwindcss.com/docs/margin
       */
      my: [{
        my: [A]
      }],
      /**
       * Margin Start
       * @see https://tailwindcss.com/docs/margin
       */
      ms: [{
        ms: [A]
      }],
      /**
       * Margin End
       * @see https://tailwindcss.com/docs/margin
       */
      me: [{
        me: [A]
      }],
      /**
       * Margin Top
       * @see https://tailwindcss.com/docs/margin
       */
      mt: [{
        mt: [A]
      }],
      /**
       * Margin Right
       * @see https://tailwindcss.com/docs/margin
       */
      mr: [{
        mr: [A]
      }],
      /**
       * Margin Bottom
       * @see https://tailwindcss.com/docs/margin
       */
      mb: [{
        mb: [A]
      }],
      /**
       * Margin Left
       * @see https://tailwindcss.com/docs/margin
       */
      ml: [{
        ml: [A]
      }],
      /**
       * Space Between X
       * @see https://tailwindcss.com/docs/space
       */
      "space-x": [{
        "space-x": [J]
      }],
      /**
       * Space Between X Reverse
       * @see https://tailwindcss.com/docs/space
       */
      "space-x-reverse": ["space-x-reverse"],
      /**
       * Space Between Y
       * @see https://tailwindcss.com/docs/space
       */
      "space-y": [{
        "space-y": [J]
      }],
      /**
       * Space Between Y Reverse
       * @see https://tailwindcss.com/docs/space
       */
      "space-y-reverse": ["space-y-reverse"],
      // Sizing
      /**
       * Width
       * @see https://tailwindcss.com/docs/width
       */
      w: [{
        w: ["auto", "min", "max", "fit", y, t]
      }],
      /**
       * Min-Width
       * @see https://tailwindcss.com/docs/min-width
       */
      "min-w": [{
        "min-w": ["min", "max", "fit", y, V]
      }],
      /**
       * Max-Width
       * @see https://tailwindcss.com/docs/max-width
       */
      "max-w": [{
        "max-w": ["0", "none", "full", "min", "max", "fit", "prose", {
          screen: [Q]
        }, Q, y]
      }],
      /**
       * Height
       * @see https://tailwindcss.com/docs/height
       */
      h: [{
        h: [y, t, "auto", "min", "max", "fit"]
      }],
      /**
       * Min-Height
       * @see https://tailwindcss.com/docs/min-height
       */
      "min-h": [{
        "min-h": ["min", "max", "fit", y, V]
      }],
      /**
       * Max-Height
       * @see https://tailwindcss.com/docs/max-height
       */
      "max-h": [{
        "max-h": [y, t, "min", "max", "fit"]
      }],
      // Typography
      /**
       * Font Size
       * @see https://tailwindcss.com/docs/font-size
       */
      "font-size": [{
        text: ["base", Q, Fr]
      }],
      /**
       * Font Smoothing
       * @see https://tailwindcss.com/docs/font-smoothing
       */
      "font-smoothing": ["antialiased", "subpixel-antialiased"],
      /**
       * Font Style
       * @see https://tailwindcss.com/docs/font-style
       */
      "font-style": ["italic", "not-italic"],
      /**
       * Font Weight
       * @see https://tailwindcss.com/docs/font-weight
       */
      "font-weight": [{
        font: ["thin", "extralight", "light", "normal", "medium", "semibold", "bold", "extrabold", "black", Tr]
      }],
      /**
       * Font Family
       * @see https://tailwindcss.com/docs/font-family
       */
      "font-family": [{
        font: [hr]
      }],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-normal": ["normal-nums"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-ordinal": ["ordinal"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-slashed-zero": ["slashed-zero"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-figure": ["lining-nums", "oldstyle-nums"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-spacing": ["proportional-nums", "tabular-nums"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-fraction": ["diagonal-fractions", "stacked-fractons"],
      /**
       * Letter Spacing
       * @see https://tailwindcss.com/docs/letter-spacing
       */
      tracking: [{
        tracking: ["tighter", "tight", "normal", "wide", "wider", "widest", y]
      }],
      /**
       * Line Clamp
       * @see https://tailwindcss.com/docs/line-clamp
       */
      "line-clamp": [{
        "line-clamp": ["none", ar, Tr]
      }],
      /**
       * Line Height
       * @see https://tailwindcss.com/docs/line-height
       */
      leading: [{
        leading: ["none", "tight", "snug", "normal", "relaxed", "loose", y, V]
      }],
      /**
       * List Style Image
       * @see https://tailwindcss.com/docs/list-style-image
       */
      "list-image": [{
        "list-image": ["none", y]
      }],
      /**
       * List Style Type
       * @see https://tailwindcss.com/docs/list-style-type
       */
      "list-style-type": [{
        list: ["none", "disc", "decimal", y]
      }],
      /**
       * List Style Position
       * @see https://tailwindcss.com/docs/list-style-position
       */
      "list-style-position": [{
        list: ["inside", "outside"]
      }],
      /**
       * Placeholder Color
       * @deprecated since Tailwind CSS v3.0.0
       * @see https://tailwindcss.com/docs/placeholder-color
       */
      "placeholder-color": [{
        placeholder: [e]
      }],
      /**
       * Placeholder Opacity
       * @see https://tailwindcss.com/docs/placeholder-opacity
       */
      "placeholder-opacity": [{
        "placeholder-opacity": [k]
      }],
      /**
       * Text Alignment
       * @see https://tailwindcss.com/docs/text-align
       */
      "text-alignment": [{
        text: ["left", "center", "right", "justify", "start", "end"]
      }],
      /**
       * Text Color
       * @see https://tailwindcss.com/docs/text-color
       */
      "text-color": [{
        text: [e]
      }],
      /**
       * Text Opacity
       * @see https://tailwindcss.com/docs/text-opacity
       */
      "text-opacity": [{
        "text-opacity": [k]
      }],
      /**
       * Text Decoration
       * @see https://tailwindcss.com/docs/text-decoration
       */
      "text-decoration": ["underline", "overline", "line-through", "no-underline"],
      /**
       * Text Decoration Style
       * @see https://tailwindcss.com/docs/text-decoration-style
       */
      "text-decoration-style": [{
        decoration: [].concat(nr(), ["wavy"])
      }],
      /**
       * Text Decoration Thickness
       * @see https://tailwindcss.com/docs/text-decoration-thickness
       */
      "text-decoration-thickness": [{
        decoration: ["auto", "from-font", V]
      }],
      /**
       * Text Underline Offset
       * @see https://tailwindcss.com/docs/text-underline-offset
       */
      "underline-offset": [{
        "underline-offset": ["auto", y, V]
      }],
      /**
       * Text Decoration Color
       * @see https://tailwindcss.com/docs/text-decoration-color
       */
      "text-decoration-color": [{
        decoration: [e]
      }],
      /**
       * Text Transform
       * @see https://tailwindcss.com/docs/text-transform
       */
      "text-transform": ["uppercase", "lowercase", "capitalize", "normal-case"],
      /**
       * Text Overflow
       * @see https://tailwindcss.com/docs/text-overflow
       */
      "text-overflow": ["truncate", "text-ellipsis", "text-clip"],
      /**
       * Text Indent
       * @see https://tailwindcss.com/docs/text-indent
       */
      indent: [{
        indent: S()
      }],
      /**
       * Vertical Alignment
       * @see https://tailwindcss.com/docs/vertical-align
       */
      "vertical-align": [{
        align: ["baseline", "top", "middle", "bottom", "text-top", "text-bottom", "sub", "super", y]
      }],
      /**
       * Whitespace
       * @see https://tailwindcss.com/docs/whitespace
       */
      whitespace: [{
        whitespace: ["normal", "nowrap", "pre", "pre-line", "pre-wrap", "break-spaces"]
      }],
      /**
       * Word Break
       * @see https://tailwindcss.com/docs/word-break
       */
      break: [{
        break: ["normal", "words", "all", "keep"]
      }],
      /**
       * Hyphens
       * @see https://tailwindcss.com/docs/hyphens
       */
      hyphens: [{
        hyphens: ["none", "manual", "auto"]
      }],
      /**
       * Content
       * @see https://tailwindcss.com/docs/content
       */
      content: [{
        content: ["none", y]
      }],
      // Backgrounds
      /**
       * Background Attachment
       * @see https://tailwindcss.com/docs/background-attachment
       */
      "bg-attachment": [{
        bg: ["fixed", "local", "scroll"]
      }],
      /**
       * Background Clip
       * @see https://tailwindcss.com/docs/background-clip
       */
      "bg-clip": [{
        "bg-clip": ["border", "padding", "content", "text"]
      }],
      /**
       * Background Opacity
       * @deprecated since Tailwind CSS v3.0.0
       * @see https://tailwindcss.com/docs/background-opacity
       */
      "bg-opacity": [{
        "bg-opacity": [k]
      }],
      /**
       * Background Origin
       * @see https://tailwindcss.com/docs/background-origin
       */
      "bg-origin": [{
        "bg-origin": ["border", "padding", "content"]
      }],
      /**
       * Background Position
       * @see https://tailwindcss.com/docs/background-position
       */
      "bg-position": [{
        bg: [].concat(X(), [Pt])
      }],
      /**
       * Background Repeat
       * @see https://tailwindcss.com/docs/background-repeat
       */
      "bg-repeat": [{
        bg: ["no-repeat", {
          repeat: ["", "x", "y", "round", "space"]
        }]
      }],
      /**
       * Background Size
       * @see https://tailwindcss.com/docs/background-size
       */
      "bg-size": [{
        bg: ["auto", "cover", "contain", kt]
      }],
      /**
       * Background Image
       * @see https://tailwindcss.com/docs/background-image
       */
      "bg-image": [{
        bg: ["none", {
          "gradient-to": ["t", "tr", "r", "br", "b", "bl", "l", "tl"]
        }, _t]
      }],
      /**
       * Background Color
       * @see https://tailwindcss.com/docs/background-color
       */
      "bg-color": [{
        bg: [e]
      }],
      /**
       * Gradient Color Stops From Position
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-from-pos": [{
        from: [R]
      }],
      /**
       * Gradient Color Stops Via Position
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-via-pos": [{
        via: [R]
      }],
      /**
       * Gradient Color Stops To Position
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-to-pos": [{
        to: [R]
      }],
      /**
       * Gradient Color Stops From
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-from": [{
        from: [x]
      }],
      /**
       * Gradient Color Stops Via
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-via": [{
        via: [x]
      }],
      /**
       * Gradient Color Stops To
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-to": [{
        to: [x]
      }],
      // Borders
      /**
       * Border Radius
       * @see https://tailwindcss.com/docs/border-radius
       */
      rounded: [{
        rounded: [c]
      }],
      /**
       * Border Radius Start
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-s": [{
        "rounded-s": [c]
      }],
      /**
       * Border Radius End
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-e": [{
        "rounded-e": [c]
      }],
      /**
       * Border Radius Top
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-t": [{
        "rounded-t": [c]
      }],
      /**
       * Border Radius Right
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-r": [{
        "rounded-r": [c]
      }],
      /**
       * Border Radius Bottom
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-b": [{
        "rounded-b": [c]
      }],
      /**
       * Border Radius Left
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-l": [{
        "rounded-l": [c]
      }],
      /**
       * Border Radius Start Start
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-ss": [{
        "rounded-ss": [c]
      }],
      /**
       * Border Radius Start End
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-se": [{
        "rounded-se": [c]
      }],
      /**
       * Border Radius End End
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-ee": [{
        "rounded-ee": [c]
      }],
      /**
       * Border Radius End Start
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-es": [{
        "rounded-es": [c]
      }],
      /**
       * Border Radius Top Left
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-tl": [{
        "rounded-tl": [c]
      }],
      /**
       * Border Radius Top Right
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-tr": [{
        "rounded-tr": [c]
      }],
      /**
       * Border Radius Bottom Right
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-br": [{
        "rounded-br": [c]
      }],
      /**
       * Border Radius Bottom Left
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-bl": [{
        "rounded-bl": [c]
      }],
      /**
       * Border Width
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w": [{
        border: [s]
      }],
      /**
       * Border Width X
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-x": [{
        "border-x": [s]
      }],
      /**
       * Border Width Y
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-y": [{
        "border-y": [s]
      }],
      /**
       * Border Width Start
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-s": [{
        "border-s": [s]
      }],
      /**
       * Border Width End
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-e": [{
        "border-e": [s]
      }],
      /**
       * Border Width Top
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-t": [{
        "border-t": [s]
      }],
      /**
       * Border Width Right
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-r": [{
        "border-r": [s]
      }],
      /**
       * Border Width Bottom
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-b": [{
        "border-b": [s]
      }],
      /**
       * Border Width Left
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-l": [{
        "border-l": [s]
      }],
      /**
       * Border Opacity
       * @see https://tailwindcss.com/docs/border-opacity
       */
      "border-opacity": [{
        "border-opacity": [k]
      }],
      /**
       * Border Style
       * @see https://tailwindcss.com/docs/border-style
       */
      "border-style": [{
        border: [].concat(nr(), ["hidden"])
      }],
      /**
       * Divide Width X
       * @see https://tailwindcss.com/docs/divide-width
       */
      "divide-x": [{
        "divide-x": [s]
      }],
      /**
       * Divide Width X Reverse
       * @see https://tailwindcss.com/docs/divide-width
       */
      "divide-x-reverse": ["divide-x-reverse"],
      /**
       * Divide Width Y
       * @see https://tailwindcss.com/docs/divide-width
       */
      "divide-y": [{
        "divide-y": [s]
      }],
      /**
       * Divide Width Y Reverse
       * @see https://tailwindcss.com/docs/divide-width
       */
      "divide-y-reverse": ["divide-y-reverse"],
      /**
       * Divide Opacity
       * @see https://tailwindcss.com/docs/divide-opacity
       */
      "divide-opacity": [{
        "divide-opacity": [k]
      }],
      /**
       * Divide Style
       * @see https://tailwindcss.com/docs/divide-style
       */
      "divide-style": [{
        divide: nr()
      }],
      /**
       * Border Color
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color": [{
        border: [a]
      }],
      /**
       * Border Color X
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-x": [{
        "border-x": [a]
      }],
      /**
       * Border Color Y
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-y": [{
        "border-y": [a]
      }],
      /**
       * Border Color Top
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-t": [{
        "border-t": [a]
      }],
      /**
       * Border Color Right
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-r": [{
        "border-r": [a]
      }],
      /**
       * Border Color Bottom
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-b": [{
        "border-b": [a]
      }],
      /**
       * Border Color Left
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-l": [{
        "border-l": [a]
      }],
      /**
       * Divide Color
       * @see https://tailwindcss.com/docs/divide-color
       */
      "divide-color": [{
        divide: [a]
      }],
      /**
       * Outline Style
       * @see https://tailwindcss.com/docs/outline-style
       */
      "outline-style": [{
        outline: [""].concat(nr())
      }],
      /**
       * Outline Offset
       * @see https://tailwindcss.com/docs/outline-offset
       */
      "outline-offset": [{
        "outline-offset": [y, V]
      }],
      /**
       * Outline Width
       * @see https://tailwindcss.com/docs/outline-width
       */
      "outline-w": [{
        outline: [V]
      }],
      /**
       * Outline Color
       * @see https://tailwindcss.com/docs/outline-color
       */
      "outline-color": [{
        outline: [e]
      }],
      /**
       * Ring Width
       * @see https://tailwindcss.com/docs/ring-width
       */
      "ring-w": [{
        ring: z()
      }],
      /**
       * Ring Width Inset
       * @see https://tailwindcss.com/docs/ring-width
       */
      "ring-w-inset": ["ring-inset"],
      /**
       * Ring Color
       * @see https://tailwindcss.com/docs/ring-color
       */
      "ring-color": [{
        ring: [e]
      }],
      /**
       * Ring Opacity
       * @see https://tailwindcss.com/docs/ring-opacity
       */
      "ring-opacity": [{
        "ring-opacity": [k]
      }],
      /**
       * Ring Offset Width
       * @see https://tailwindcss.com/docs/ring-offset-width
       */
      "ring-offset-w": [{
        "ring-offset": [V]
      }],
      /**
       * Ring Offset Color
       * @see https://tailwindcss.com/docs/ring-offset-color
       */
      "ring-offset-color": [{
        "ring-offset": [e]
      }],
      // Effects
      /**
       * Box Shadow
       * @see https://tailwindcss.com/docs/box-shadow
       */
      shadow: [{
        shadow: ["", "inner", "none", Q, At]
      }],
      /**
       * Box Shadow Color
       * @see https://tailwindcss.com/docs/box-shadow-color
       */
      "shadow-color": [{
        shadow: [hr]
      }],
      /**
       * Opacity
       * @see https://tailwindcss.com/docs/opacity
       */
      opacity: [{
        opacity: [k]
      }],
      /**
       * Mix Blend Mode
       * @see https://tailwindcss.com/docs/mix-blend-mode
       */
      "mix-blend": [{
        "mix-blend": fr()
      }],
      /**
       * Background Blend Mode
       * @see https://tailwindcss.com/docs/background-blend-mode
       */
      "bg-blend": [{
        "bg-blend": fr()
      }],
      // Filters
      /**
       * Filter
       * @deprecated since Tailwind CSS v3.0.0
       * @see https://tailwindcss.com/docs/filter
       */
      filter: [{
        filter: ["", "none"]
      }],
      /**
       * Blur
       * @see https://tailwindcss.com/docs/blur
       */
      blur: [{
        blur: [o]
      }],
      /**
       * Brightness
       * @see https://tailwindcss.com/docs/brightness
       */
      brightness: [{
        brightness: [n]
      }],
      /**
       * Contrast
       * @see https://tailwindcss.com/docs/contrast
       */
      contrast: [{
        contrast: [u]
      }],
      /**
       * Drop Shadow
       * @see https://tailwindcss.com/docs/drop-shadow
       */
      "drop-shadow": [{
        "drop-shadow": ["", "none", Q, y]
      }],
      /**
       * Grayscale
       * @see https://tailwindcss.com/docs/grayscale
       */
      grayscale: [{
        grayscale: [v]
      }],
      /**
       * Hue Rotate
       * @see https://tailwindcss.com/docs/hue-rotate
       */
      "hue-rotate": [{
        "hue-rotate": [p]
      }],
      /**
       * Invert
       * @see https://tailwindcss.com/docs/invert
       */
      invert: [{
        invert: [f]
      }],
      /**
       * Saturate
       * @see https://tailwindcss.com/docs/saturate
       */
      saturate: [{
        saturate: [_]
      }],
      /**
       * Sepia
       * @see https://tailwindcss.com/docs/sepia
       */
      sepia: [{
        sepia: [N]
      }],
      /**
       * Backdrop Filter
       * @deprecated since Tailwind CSS v3.0.0
       * @see https://tailwindcss.com/docs/backdrop-filter
       */
      "backdrop-filter": [{
        "backdrop-filter": ["", "none"]
      }],
      /**
       * Backdrop Blur
       * @see https://tailwindcss.com/docs/backdrop-blur
       */
      "backdrop-blur": [{
        "backdrop-blur": [o]
      }],
      /**
       * Backdrop Brightness
       * @see https://tailwindcss.com/docs/backdrop-brightness
       */
      "backdrop-brightness": [{
        "backdrop-brightness": [n]
      }],
      /**
       * Backdrop Contrast
       * @see https://tailwindcss.com/docs/backdrop-contrast
       */
      "backdrop-contrast": [{
        "backdrop-contrast": [u]
      }],
      /**
       * Backdrop Grayscale
       * @see https://tailwindcss.com/docs/backdrop-grayscale
       */
      "backdrop-grayscale": [{
        "backdrop-grayscale": [v]
      }],
      /**
       * Backdrop Hue Rotate
       * @see https://tailwindcss.com/docs/backdrop-hue-rotate
       */
      "backdrop-hue-rotate": [{
        "backdrop-hue-rotate": [p]
      }],
      /**
       * Backdrop Invert
       * @see https://tailwindcss.com/docs/backdrop-invert
       */
      "backdrop-invert": [{
        "backdrop-invert": [f]
      }],
      /**
       * Backdrop Opacity
       * @see https://tailwindcss.com/docs/backdrop-opacity
       */
      "backdrop-opacity": [{
        "backdrop-opacity": [k]
      }],
      /**
       * Backdrop Saturate
       * @see https://tailwindcss.com/docs/backdrop-saturate
       */
      "backdrop-saturate": [{
        "backdrop-saturate": [_]
      }],
      /**
       * Backdrop Sepia
       * @see https://tailwindcss.com/docs/backdrop-sepia
       */
      "backdrop-sepia": [{
        "backdrop-sepia": [N]
      }],
      // Tables
      /**
       * Border Collapse
       * @see https://tailwindcss.com/docs/border-collapse
       */
      "border-collapse": [{
        border: ["collapse", "separate"]
      }],
      /**
       * Border Spacing
       * @see https://tailwindcss.com/docs/border-spacing
       */
      "border-spacing": [{
        "border-spacing": [i]
      }],
      /**
       * Border Spacing X
       * @see https://tailwindcss.com/docs/border-spacing
       */
      "border-spacing-x": [{
        "border-spacing-x": [i]
      }],
      /**
       * Border Spacing Y
       * @see https://tailwindcss.com/docs/border-spacing
       */
      "border-spacing-y": [{
        "border-spacing-y": [i]
      }],
      /**
       * Table Layout
       * @see https://tailwindcss.com/docs/table-layout
       */
      "table-layout": [{
        table: ["auto", "fixed"]
      }],
      /**
       * Caption Side
       * @see https://tailwindcss.com/docs/caption-side
       */
      caption: [{
        caption: ["top", "bottom"]
      }],
      // Transitions and Animation
      /**
       * Tranisition Property
       * @see https://tailwindcss.com/docs/transition-property
       */
      transition: [{
        transition: ["none", "all", "", "colors", "opacity", "shadow", "transform", y]
      }],
      /**
       * Transition Duration
       * @see https://tailwindcss.com/docs/transition-duration
       */
      duration: [{
        duration: or()
      }],
      /**
       * Transition Timing Function
       * @see https://tailwindcss.com/docs/transition-timing-function
       */
      ease: [{
        ease: ["linear", "in", "out", "in-out", y]
      }],
      /**
       * Transition Delay
       * @see https://tailwindcss.com/docs/transition-delay
       */
      delay: [{
        delay: or()
      }],
      /**
       * Animation
       * @see https://tailwindcss.com/docs/animation
       */
      animate: [{
        animate: ["none", "spin", "ping", "pulse", "bounce", y]
      }],
      // Transforms
      /**
       * Transform
       * @see https://tailwindcss.com/docs/transform
       */
      transform: [{
        transform: ["", "gpu", "none"]
      }],
      /**
       * Scale
       * @see https://tailwindcss.com/docs/scale
       */
      scale: [{
        scale: [M]
      }],
      /**
       * Scale X
       * @see https://tailwindcss.com/docs/scale
       */
      "scale-x": [{
        "scale-x": [M]
      }],
      /**
       * Scale Y
       * @see https://tailwindcss.com/docs/scale
       */
      "scale-y": [{
        "scale-y": [M]
      }],
      /**
       * Rotate
       * @see https://tailwindcss.com/docs/rotate
       */
      rotate: [{
        rotate: [mr, y]
      }],
      /**
       * Translate X
       * @see https://tailwindcss.com/docs/translate
       */
      "translate-x": [{
        "translate-x": [Y]
      }],
      /**
       * Translate Y
       * @see https://tailwindcss.com/docs/translate
       */
      "translate-y": [{
        "translate-y": [Y]
      }],
      /**
       * Skew X
       * @see https://tailwindcss.com/docs/skew
       */
      "skew-x": [{
        "skew-x": [L]
      }],
      /**
       * Skew Y
       * @see https://tailwindcss.com/docs/skew
       */
      "skew-y": [{
        "skew-y": [L]
      }],
      /**
       * Transform Origin
       * @see https://tailwindcss.com/docs/transform-origin
       */
      "transform-origin": [{
        origin: ["center", "top", "top-right", "right", "bottom-right", "bottom", "bottom-left", "left", "top-left", y]
      }],
      // Interactivity
      /**
       * Accent Color
       * @see https://tailwindcss.com/docs/accent-color
       */
      accent: [{
        accent: ["auto", e]
      }],
      /**
       * Appearance
       * @see https://tailwindcss.com/docs/appearance
       */
      appearance: ["appearance-none"],
      /**
       * Cursor
       * @see https://tailwindcss.com/docs/cursor
       */
      cursor: [{
        cursor: ["auto", "default", "pointer", "wait", "text", "move", "help", "not-allowed", "none", "context-menu", "progress", "cell", "crosshair", "vertical-text", "alias", "copy", "no-drop", "grab", "grabbing", "all-scroll", "col-resize", "row-resize", "n-resize", "e-resize", "s-resize", "w-resize", "ne-resize", "nw-resize", "se-resize", "sw-resize", "ew-resize", "ns-resize", "nesw-resize", "nwse-resize", "zoom-in", "zoom-out", y]
      }],
      /**
       * Caret Color
       * @see https://tailwindcss.com/docs/just-in-time-mode#caret-color-utilities
       */
      "caret-color": [{
        caret: [e]
      }],
      /**
       * Pointer Events
       * @see https://tailwindcss.com/docs/pointer-events
       */
      "pointer-events": [{
        "pointer-events": ["none", "auto"]
      }],
      /**
       * Resize
       * @see https://tailwindcss.com/docs/resize
       */
      resize: [{
        resize: ["none", "y", "x", ""]
      }],
      /**
       * Scroll Behavior
       * @see https://tailwindcss.com/docs/scroll-behavior
       */
      "scroll-behavior": [{
        scroll: ["auto", "smooth"]
      }],
      /**
       * Scroll Margin
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-m": [{
        "scroll-m": S()
      }],
      /**
       * Scroll Margin X
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mx": [{
        "scroll-mx": S()
      }],
      /**
       * Scroll Margin Y
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-my": [{
        "scroll-my": S()
      }],
      /**
       * Scroll Margin Start
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-ms": [{
        "scroll-ms": S()
      }],
      /**
       * Scroll Margin End
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-me": [{
        "scroll-me": S()
      }],
      /**
       * Scroll Margin Top
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mt": [{
        "scroll-mt": S()
      }],
      /**
       * Scroll Margin Right
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mr": [{
        "scroll-mr": S()
      }],
      /**
       * Scroll Margin Bottom
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mb": [{
        "scroll-mb": S()
      }],
      /**
       * Scroll Margin Left
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-ml": [{
        "scroll-ml": S()
      }],
      /**
       * Scroll Padding
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-p": [{
        "scroll-p": S()
      }],
      /**
       * Scroll Padding X
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-px": [{
        "scroll-px": S()
      }],
      /**
       * Scroll Padding Y
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-py": [{
        "scroll-py": S()
      }],
      /**
       * Scroll Padding Start
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-ps": [{
        "scroll-ps": S()
      }],
      /**
       * Scroll Padding End
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pe": [{
        "scroll-pe": S()
      }],
      /**
       * Scroll Padding Top
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pt": [{
        "scroll-pt": S()
      }],
      /**
       * Scroll Padding Right
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pr": [{
        "scroll-pr": S()
      }],
      /**
       * Scroll Padding Bottom
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pb": [{
        "scroll-pb": S()
      }],
      /**
       * Scroll Padding Left
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pl": [{
        "scroll-pl": S()
      }],
      /**
       * Scroll Snap Align
       * @see https://tailwindcss.com/docs/scroll-snap-align
       */
      "snap-align": [{
        snap: ["start", "end", "center", "align-none"]
      }],
      /**
       * Scroll Snap Stop
       * @see https://tailwindcss.com/docs/scroll-snap-stop
       */
      "snap-stop": [{
        snap: ["normal", "always"]
      }],
      /**
       * Scroll Snap Type
       * @see https://tailwindcss.com/docs/scroll-snap-type
       */
      "snap-type": [{
        snap: ["none", "x", "y", "both"]
      }],
      /**
       * Scroll Snap Type Strictness
       * @see https://tailwindcss.com/docs/scroll-snap-type
       */
      "snap-strictness": [{
        snap: ["mandatory", "proximity"]
      }],
      /**
       * Touch Action
       * @see https://tailwindcss.com/docs/touch-action
       */
      touch: [{
        touch: ["auto", "none", "pinch-zoom", "manipulation", {
          pan: ["x", "left", "right", "y", "up", "down"]
        }]
      }],
      /**
       * User Select
       * @see https://tailwindcss.com/docs/user-select
       */
      select: [{
        select: ["none", "text", "all", "auto"]
      }],
      /**
       * Will Change
       * @see https://tailwindcss.com/docs/will-change
       */
      "will-change": [{
        "will-change": ["auto", "scroll", "contents", "transform", y]
      }],
      // SVG
      /**
       * Fill
       * @see https://tailwindcss.com/docs/fill
       */
      fill: [{
        fill: [e, "none"]
      }],
      /**
       * Stroke Width
       * @see https://tailwindcss.com/docs/stroke-width
       */
      "stroke-w": [{
        stroke: [V, Tr]
      }],
      /**
       * Stroke
       * @see https://tailwindcss.com/docs/stroke
       */
      stroke: [{
        stroke: [e, "none"]
      }],
      // Accessibility
      /**
       * Screen Readers
       * @see https://tailwindcss.com/docs/screen-readers
       */
      sr: ["sr-only", "not-sr-only"]
    },
    conflictingClassGroups: {
      overflow: ["overflow-x", "overflow-y"],
      overscroll: ["overscroll-x", "overscroll-y"],
      inset: ["inset-x", "inset-y", "start", "end", "top", "right", "bottom", "left"],
      "inset-x": ["right", "left"],
      "inset-y": ["top", "bottom"],
      flex: ["basis", "grow", "shrink"],
      gap: ["gap-x", "gap-y"],
      p: ["px", "py", "ps", "pe", "pt", "pr", "pb", "pl"],
      px: ["pr", "pl"],
      py: ["pt", "pb"],
      m: ["mx", "my", "ms", "me", "mt", "mr", "mb", "ml"],
      mx: ["mr", "ml"],
      my: ["mt", "mb"],
      "font-size": ["leading"],
      "fvn-normal": ["fvn-ordinal", "fvn-slashed-zero", "fvn-figure", "fvn-spacing", "fvn-fraction"],
      "fvn-ordinal": ["fvn-normal"],
      "fvn-slashed-zero": ["fvn-normal"],
      "fvn-figure": ["fvn-normal"],
      "fvn-spacing": ["fvn-normal"],
      "fvn-fraction": ["fvn-normal"],
      rounded: ["rounded-s", "rounded-e", "rounded-t", "rounded-r", "rounded-b", "rounded-l", "rounded-ss", "rounded-se", "rounded-ee", "rounded-es", "rounded-tl", "rounded-tr", "rounded-br", "rounded-bl"],
      "rounded-s": ["rounded-ss", "rounded-es"],
      "rounded-e": ["rounded-se", "rounded-ee"],
      "rounded-t": ["rounded-tl", "rounded-tr"],
      "rounded-r": ["rounded-tr", "rounded-br"],
      "rounded-b": ["rounded-br", "rounded-bl"],
      "rounded-l": ["rounded-tl", "rounded-bl"],
      "border-spacing": ["border-spacing-x", "border-spacing-y"],
      "border-w": ["border-w-s", "border-w-e", "border-w-t", "border-w-r", "border-w-b", "border-w-l"],
      "border-w-x": ["border-w-r", "border-w-l"],
      "border-w-y": ["border-w-t", "border-w-b"],
      "border-color": ["border-color-t", "border-color-r", "border-color-b", "border-color-l"],
      "border-color-x": ["border-color-r", "border-color-l"],
      "border-color-y": ["border-color-t", "border-color-b"],
      "scroll-m": ["scroll-mx", "scroll-my", "scroll-ms", "scroll-me", "scroll-mt", "scroll-mr", "scroll-mb", "scroll-ml"],
      "scroll-mx": ["scroll-mr", "scroll-ml"],
      "scroll-my": ["scroll-mt", "scroll-mb"],
      "scroll-p": ["scroll-px", "scroll-py", "scroll-ps", "scroll-pe", "scroll-pt", "scroll-pr", "scroll-pb", "scroll-pl"],
      "scroll-px": ["scroll-pr", "scroll-pl"],
      "scroll-py": ["scroll-pt", "scroll-pb"]
    },
    conflictingClassGroupModifiers: {
      "font-size": ["leading"]
    }
  };
}
var zt = /* @__PURE__ */ wt($t);
function xr(...e) {
  return zt(lt(e));
}
function Se(e, t) {
  return rr.Children.toArray(e).filter((o) => rr.isValidElement(o) && o.type === t);
}
function Nt(e, t) {
  let o = 0;
  return rr.Children.forEach(e, (n) => {
    if (rr.isValidElement(n)) {
      const a = Se(n.props.children, t);
      o += a.length;
    }
  }), o;
}
function Pr() {
  return Pr = Object.assign ? Object.assign.bind() : function(e) {
    for (var t = 1; t < arguments.length; t++) {
      var o = arguments[t];
      for (var n in o)
        Object.prototype.hasOwnProperty.call(o, n) && (e[n] = o[n]);
    }
    return e;
  }, Pr.apply(this, arguments);
}
function Wt(e, t) {
  typeof e == "function" ? e(t) : e != null && (e.current = t);
}
function Gt(...e) {
  return (t) => e.forEach(
    (o) => Wt(o, t)
  );
}
const sr = /* @__PURE__ */ D((e, t) => {
  const { children: o, ...n } = e, a = yr.toArray(o), c = a.find(Ft);
  if (c) {
    const i = c.props.children, s = a.map((u) => u === c ? yr.count(i) > 1 ? yr.only(null) : /* @__PURE__ */ kr(i) ? i.props.children : null : u);
    return /* @__PURE__ */ Wr(Br, Pr({}, n, {
      ref: t
    }), /* @__PURE__ */ kr(i) ? /* @__PURE__ */ be(i, void 0, s) : null);
  }
  return /* @__PURE__ */ Wr(Br, Pr({}, n, {
    ref: t
  }), o);
});
sr.displayName = "Slot";
const Br = /* @__PURE__ */ D((e, t) => {
  const { children: o, ...n } = e;
  return /* @__PURE__ */ kr(o) ? /* @__PURE__ */ be(o, {
    ...Bt(n, o.props),
    ref: t ? Gt(t, o.ref) : o.ref
  }) : yr.count(o) > 1 ? yr.only(null) : null;
});
Br.displayName = "SlotClone";
const Lt = ({ children: e }) => /* @__PURE__ */ Wr(ot, null, e);
function Ft(e) {
  return /* @__PURE__ */ kr(e) && e.type === Lt;
}
function Bt(e, t) {
  const o = {
    ...t
  };
  for (const n in t) {
    const a = e[n], c = t[n];
    /^on[A-Z]/.test(n) ? a && c ? o[n] = (...s) => {
      c(...s), a(...s);
    } : a && (o[n] = a) : n === "style" ? o[n] = {
      ...a,
      ...c
    } : n === "className" && (o[n] = [
      a,
      c
    ].filter(Boolean).join(" "));
  }
  return {
    ...e,
    ...o
  };
}
const _r = D(
  ({ className: e, children: t, asChild: o, ...n }, a) => {
    const c = o ? sr : "button";
    return /* @__PURE__ */ E.jsx(
      c,
      {
        className: xr(
          "arc-flex arc-items-center arc-justify-center arc-transition arc-duration-150 arc-ease-in-out arc-rounded-full",
          e
        ),
        ref: a,
        ...n,
        children: t
      }
    );
  }
);
_r.displayName = "Button";
const Te = D(
  ({ onClick: e, children: t, ...o }, n) => {
    const { next: a, orientation: c, hasLoop: i, currentPage: s, pagesCount: u } = q(H), v = /* @__PURE__ */ E.jsx(
      "svg",
      {
        xmlns: "http://www.w3.org/2000/svg",
        fill: "none",
        viewBox: "0 0 24 24",
        strokeWidth: 1.5,
        stroke: "currentColor",
        className: "arc-w-6 arc-h-6",
        children: /* @__PURE__ */ E.jsx(
          "path",
          {
            strokeLinecap: "round",
            strokeLinejoin: "round",
            d: "M4.5 12h15m0 0l-6.75-6.75M19.5 12l-6.75 6.75"
          }
        )
      }
    ), p = /* @__PURE__ */ E.jsx(
      "svg",
      {
        xmlns: "http://www.w3.org/2000/svg",
        fill: "none",
        viewBox: "0 0 24 24",
        strokeWidth: 1.5,
        stroke: "currentColor",
        className: "arc-w-6 arc-h-6",
        children: /* @__PURE__ */ E.jsx(
          "path",
          {
            strokeLinecap: "round",
            strokeLinejoin: "round",
            d: "M12 4.5v15m0 0l6.75-6.75M12 19.5l-6.75-6.75"
          }
        )
      }
    ), f = (g) => {
      a(), e == null || e(g);
    };
    return /* @__PURE__ */ E.jsx(
      _r,
      {
        onClick: f,
        disabled: !i && s === u - 1,
        ref: n,
        "aria-label": "Next page",
        ...o,
        children: t ?? (c === "vertical" ? p : v)
      }
    );
  }
);
Te.displayName = "CarouselNextButton";
const Yr = D(
  ({ index: e, className: t, onFocus: o, onBlur: n, onKeyDown: a, ...c }, i) => {
    const { currentPage: s, setCurrentPage: u, startAutoplay: v, stopAutoplay: p, autoplay: f, isPaused: g } = q(H), { onPageChange: x } = Dr(), R = e === s, T = () => {
      const _ = s, M = e;
      x == null || x(_, M), u(M);
    }, A = (_) => {
      f && !g && p(), o == null || o(_);
    }, k = (_) => {
      f && !g && v(), n == null || n(_);
    }, m = (_) => {
      if (_.key === "ArrowUp" || _.key === "ArrowDown") {
        _.preventDefault();
        return;
      }
      a == null || a(_);
    };
    return /* @__PURE__ */ E.jsx(E.Fragment, { children: /* @__PURE__ */ E.jsx(
      "input",
      {
        type: "radio",
        name: "arc-carousel-pagination",
        className: t,
        onChange: T,
        checked: R,
        ref: i,
        onFocus: A,
        onBlur: k,
        onKeyDown: m,
        ...c
      }
    ) });
  }
);
Yr.displayName = "CarouselPaginationItem";
const ke = D(
  ({ className: e, asChild: t, render: o, ...n }, a) => {
    const { pagesCount: c, orientation: i, currentPage: s } = q(H);
    if (c <= 1)
      return null;
    const u = t ? sr : "div";
    return /* @__PURE__ */ E.jsx(
      u,
      {
        className: xr(
          "arc-flex",
          i === "vertical" ? "arc-flex-col" : "arc-flex-row",
          e
        ),
        ref: a,
        ...n,
        children: Array.from({ length: c }).map(
          (v, p) => o ? o(p, p === s) : /* @__PURE__ */ E.jsx(Yr, { index: p }, p)
        )
      }
    );
  }
);
ke.displayName = "CarouselPagination";
const Pe = D(
  ({ onClick: e, children: t, ...o }, n) => {
    const { startAutoplay: a, stopAutoplay: c, isPaused: i, setIsPaused: s, autoplay: u } = q(H);
    if (!u)
      return null;
    const v = /* @__PURE__ */ E.jsx(
      "svg",
      {
        xmlns: "http://www.w3.org/2000/svg",
        viewBox: "0 0 24 24",
        fill: "currentColor",
        className: "arc-w-6 arc-h-6",
        children: /* @__PURE__ */ E.jsx(
          "path",
          {
            fillRule: "evenodd",
            d: "M4.5 5.653c0-1.426 1.529-2.33 2.779-1.643l11.54 6.348c1.295.712 1.295 2.573 0 3.285L7.28 19.991c-1.25.687-2.779-.217-2.779-1.643V5.653z",
            clipRule: "evenodd"
          }
        )
      }
    ), p = /* @__PURE__ */ E.jsx(
      "svg",
      {
        xmlns: "http://www.w3.org/2000/svg",
        viewBox: "0 0 24 24",
        fill: "currentColor",
        className: "arc-w-6 arc-h-6",
        children: /* @__PURE__ */ E.jsx(
          "path",
          {
            fillRule: "evenodd",
            d: "M6.75 5.25a.75.75 0 01.75-.75H9a.75.75 0 01.75.75v13.5a.75.75 0 01-.75.75H7.5a.75.75 0 01-.75-.75V5.25zm7.5 0A.75.75 0 0115 4.5h1.5a.75.75 0 01.75.75v13.5a.75.75 0 01-.75.75H15a.75.75 0 01-.75-.75V5.25z",
            clipRule: "evenodd"
          }
        )
      }
    ), f = (g) => {
      i ? (s(!1), a()) : (s(!0), c()), e == null || e(g);
    };
    return /* @__PURE__ */ E.jsx(
      _r,
      {
        onClick: f,
        ref: n,
        "aria-label": i ? "Start slide autoplay" : "Pause slide autoplay",
        ...o,
        children: t ?? (i ? v : p)
      }
    );
  }
);
Pe.displayName = "CarouselPlayPauseButton";
const _e = D(
  ({ onClick: e, children: t, ...o }, n) => {
    const { prev: a, orientation: c, hasLoop: i, currentPage: s } = q(H), u = /* @__PURE__ */ E.jsx(
      "svg",
      {
        xmlns: "http://www.w3.org/2000/svg",
        fill: "none",
        viewBox: "0 0 24 24",
        strokeWidth: 1.5,
        stroke: "currentColor",
        className: "arc-w-6 arc-h-6",
        children: /* @__PURE__ */ E.jsx(
          "path",
          {
            strokeLinecap: "round",
            strokeLinejoin: "round",
            d: "M19.5 12h-15m0 0l6.75 6.75M4.5 12l6.75-6.75"
          }
        )
      }
    ), v = /* @__PURE__ */ E.jsx(
      "svg",
      {
        xmlns: "http://www.w3.org/2000/svg",
        fill: "none",
        viewBox: "0 0 24 24",
        strokeWidth: 1.5,
        stroke: "currentColor",
        className: "arc-w-6 arc-h-6",
        children: /* @__PURE__ */ E.jsx(
          "path",
          {
            strokeLinecap: "round",
            strokeLinejoin: "round",
            d: "M12 19.5v-15m0 0l-6.75 6.75M12 4.5l6.75 6.75"
          }
        )
      }
    ), p = (f) => {
      a(), e == null || e(f);
    };
    return /* @__PURE__ */ E.jsx(
      _r,
      {
        onClick: p,
        disabled: !i && s === 0,
        ref: n,
        "aria-label": "Previous page",
        ...o,
        children: t ?? (c === "vertical" ? v : u)
      }
    );
  }
);
_e.displayName = "CarouselPrevButton";
const je = D(
  ({ asChild: e, render: t, className: o, ...n }, a) => {
    const { pagesCount: c, currentPage: i } = q(H);
    if (c <= 1)
      return null;
    const s = (i + 1) / c * 100, u = e ? sr : "div";
    return /* @__PURE__ */ E.jsx(
      u,
      {
        className: xr("arc-flex-grow arc-h-1 arc-rounded-full arc-overflow-hidden", o),
        ref: a,
        ...n,
        children: t ? t(s) : /* @__PURE__ */ E.jsx(
          "progress",
          {
            className: "arc-h-full arc-bg-black arc-transition-[width] arc-duration-150 arc-ease-in-out",
            style: { width: `${s}%` }
          }
        )
      }
    );
  }
);
je.displayName = "CarouselProgress";
function Dt({
  orientation: e,
  currentPage: t,
  slidesPerPage: o,
  slideIndex: n,
  slidePercentage: a,
  slidesGap: c,
  isActive: i
}) {
  const s = `calc(${n % o * a}% - ${c}px)`, u = e === "horizontal" ? "Y" : "X", v = Math.floor(n / o), p = v < t, f = v > t;
  return {
    position: "absolute",
    top: e === "horizontal" ? 0 : s,
    left: e === "horizontal" ? s : 0,
    backfaceVisibility: "hidden",
    transform: `rotate${u}(${p ? 180 : f ? -180 : 0}deg)`,
    zIndex: i ? 1 : 0
  };
}
const Vt = {
  perspective: 1e3,
  transformStyle: "preserve-3d",
  overflow: "visible"
}, Ur = D(
  ({ children: e, asChild: t, className: o, onMouseEnter: n, onMouseLeave: a, style: c, ...i }, s) => {
    const {
      transition: u,
      orientation: v,
      gap: p,
      isAutoplaying: f,
      isPaused: g,
      stopAutoplay: x,
      startAutoplay: R,
      autoplay: T
    } = q(H), A = t ? sr : "div", k = rr.Children.map(e, (N, L) => rr.cloneElement(N, {
      key: L,
      slideIndex: L
    })), m = (N) => {
      T && f && !g && x(), n == null || n(N);
    }, _ = (N) => {
      T && !f && !g && R(), a == null || a(N);
    };
    let M = {};
    return u === "flip" ? M = {
      ...Vt,
      ...c,
      gap: `${p}px`
    } : M = {
      ...c,
      gap: `${p}px`
    }, /* @__PURE__ */ E.jsx(
      A,
      {
        className: xr(
          "arc-relative arc-flex arc-w-full arc-h-full arc-overflow-hidden",
          v === "vertical" ? "arc-flex-col" : "arc-flex-row",
          o
        ),
        style: M,
        ref: s,
        role: "region",
        "aria-atomic": "false",
        "aria-live": f ? "off" : "polite",
        onMouseEnter: m,
        onMouseLeave: _,
        ...i,
        children: k
      }
    );
  }
);
Ur.displayName = "CarouselWrapper";
function Yt({
  slideIndex: e,
  slidePercentage: t,
  slidesPerPage: o,
  slidesGap: n,
  isActive: a,
  orientation: c
}) {
  const i = `calc(${e % o * t}% - ${n}px)`;
  return {
    position: "absolute",
    top: c === "horizontal" ? 0 : i,
    left: c === "horizontal" ? i : 0,
    opacity: a ? 1 : 0,
    zIndex: a ? 1 : 0
  };
}
function Ut({
  orientation: e,
  gap: t,
  currentPage: o,
  slidesPerPage: n
}) {
  return {
    transform: `translate${e === "horizontal" ? "X" : "Y"}(calc(${o * n * -100}% - ${t * o * n}px))`
  };
}
const qt = {
  slide: Ut,
  fade: Yt,
  flip: Dt
};
function Ht({
  orientation: e,
  transition: t,
  gap: o,
  currentPage: n,
  slidesPerPage: a,
  slideIndex: c,
  isActive: i
}) {
  const s = 100 / a, u = (a * o - o) / a, v = qt[t], p = v({
    orientation: e,
    gap: o,
    currentPage: n,
    slidesPerPage: a,
    slidePercentage: s,
    slidesGap: u,
    slideIndex: c,
    isActive: i
  });
  return e === "horizontal" ? {
    width: `calc(${s}% - ${u}px)`,
    ...p
  } : {
    height: `calc(${s}% - ${u}px)`,
    ...p
  };
}
function Jt({
  onSwipedToPrev: e,
  onSwipedToNext: t,
  orientation: o
}) {
  const [n, a] = dr(0), [c, i] = dr(0), s = 50;
  return {
    onTouchStart: (f) => {
      i(0), a(o === "horizontal" ? f.targetTouches[0].clientX : f.targetTouches[0].clientY);
    },
    onTouchMove: (f) => {
      i(o === "horizontal" ? f.targetTouches[0].clientX : f.targetTouches[0].clientY);
    },
    onTouchEnd: () => {
      if (!n || !c)
        return;
      const f = n - c;
      if (o === "horizontal") {
        const g = f > s, x = f < -s;
        g && t(), x && e();
      } else {
        const g = f > s;
        f < -s && e(), g && t();
      }
    }
  };
}
const qr = D(
  ({ slideIndex: e = 0, asChild: t, className: o, onClick: n, onFocus: a, style: c, ...i }, s) => {
    const {
      slideCount: u,
      slidesPerPage: v,
      currentPage: p,
      orientation: f,
      transition: g,
      transitionDuration: x,
      gap: R,
      next: T,
      prev: A,
      setCurrentPage: k
    } = q(H), { onSlideClick: m, onPageChangeEnd: _ } = Dr(), M = p * v, N = (p + 1) * v - 1, L = e >= M && e <= N, J = Ht({
      gap: R,
      slidesPerPage: v,
      currentPage: p,
      orientation: f,
      transition: g,
      slideIndex: e,
      isActive: L
    }), Y = t ? sr : "div", $ = (z) => {
      n == null || n(z), m == null || m(e, z);
    }, er = Jt({
      onSwipedToNext: T,
      onSwipedToPrev: A,
      orientation: f
    }), tr = () => {
      L && (_ == null || _(p));
    }, S = (z) => {
      L || k(Math.floor(e / v)), a == null || a(z);
    };
    return /* @__PURE__ */ E.jsx(
      Y,
      {
        className: xr("arc-h-full arc-flex-shrink-0 arc-transition", o),
        style: { ...c, ...J, transitionDuration: `${x}ms` },
        ref: s,
        "data-arc-index": e,
        "data-arc-is-active": L,
        onClick: $,
        onTransitionEnd: tr,
        onFocus: S,
        role: "group",
        "aria-roledescription": "slide",
        "aria-label": `${e + 1} of ${u}`,
        ...er,
        ...i
      }
    );
  }
);
qr.displayName = "CarouselSlide";
const Xt = {
  base: 0,
  sm: 640,
  md: 768,
  lg: 1024,
  xl: 1280,
  "2xl": 1536
};
function Zt(e) {
  const [t, o] = dr("base");
  return ge(() => {
    const n = () => {
      const a = window.innerWidth, c = Object.keys(e).reverse().find((i) => a >= Xt[i]);
      o(c);
    };
    return n(), window.addEventListener("resize", n), () => {
      window.removeEventListener("resize", n);
    };
  }, []), e[t];
}
const Ae = D(
  ({
    orientation: e = "horizontal",
    slidesPerPage: t = 1,
    initialPage: o = 0,
    hasLoop: n = !1,
    transition: a = "slide",
    transitionDuration: c = 300,
    gap: i = 0,
    children: s,
    asChild: u = !1,
    onSlideClick: v,
    onPageChange: p,
    onPageChangeEnd: f,
    autoplay: g = !1,
    autoplayInterval: x = 3e3,
    ...R
  }, T) => {
    const A = u ? sr : "section", k = Se(s, Ur);
    if (k.length !== 1)
      throw new Error("CarouselRoot must have only one CarouselWrapper child");
    const m = k[0], _ = Nt(m, qr);
    typeof t != "number" && (t = Zt(t));
    const M = Math.ceil(_ / t);
    return /* @__PURE__ */ E.jsx(
      st,
      {
        onSlideClick: v,
        onPageChange: p,
        onPageChangeEnd: f,
        children: /* @__PURE__ */ E.jsx(
          he,
          {
            orientation: e,
            slideCount: _,
            pagesCount: M,
            slidesPerPage: t,
            initialPage: o,
            hasLoop: n,
            transition: a,
            transitionDuration: c,
            gap: i,
            autoplay: g,
            autoplayInterval: x,
            children: /* @__PURE__ */ E.jsx(A, { ref: T, "aria-roledescription": "carousel", ...R, children: s })
          }
        )
      }
    );
  }
);
Ae.displayName = "CarouselRoot";
const Qt = {
  Root: Ae,
  PrevButton: _e,
  NextButton: Te,
  PlayPauseButton: Pe,
  Pagination: ke,
  PaginationItem: Yr,
  Wrapper: Ur,
  Slide: qr,
  Progress: je
};
export {
  Qt as Carousel
};
